drawing_agent/main.py

```
@@ -27,110 +27,109 @@ class AnalysisRequest(BaseModel):
-        logger.error(f"Ошибка в логике агента: {error_msg}")
-        raise HTTPException(status_code=500, detail=error_msg)
+        raise HTTPException(status_code=500, detail=result.get("error"))
 
-    except HTTPException:
-        # Re-raise HTTPException without modification
-        raise
     except Exception as e:
-        logger.exception(f"Критический сбой при обработке: {str(e)}")
-        raise HTTPException(status_code=500, detail=f"Internal Server Error: {str(e)}")
+        logger.exception(e)
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)